nullaway/src/test/java/com/uber/nullaway/jspecify/JSpecifyVarargsTests.java

```diff
@@ -0,0 +1,432 @@
+package com.uber.nullaway.jspecify;
+
+import com.google.errorprone.CompilationTestHelper;
+import com.uber.nullaway.NullAwayTestsBase;
+import com.uber.nullaway.VarargsTests;
+import java.util.Arrays;
+import org.junit.Test;
+
+/**
+ * Tests for handling of varargs annotations in JSpecify mode. Based on {@link VarargsTests}, with
+ * tests and assertions modified appropriately.
+ */
+public class JSpecifyVarargsTests extends NullAwayTestsBase {
+
+  @Test
+  public void testNonNullVarargs() {
+    makeHelper()
+        .addSourceLines(
+            "Utilities.java",
+            "package com.uber;",
+            "import javax.annotation.Nullable;",
+            "public class Utilities {",
+            " public static String takesNonNullVarargs(Object o, Object... others) {",
+            "  String s = o.toString() + \" \";",
+            "  for (Object other : others) {",
+            "    s += other.toString() + \" \";",
+            "  }",
+            "  return s;",
+            " }",
+            "}")
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "import javax.annotation.Nullable;",
+            "public class Test {",
+            "  public void testNonNullVarargs(Object o1, Object o2, Object o3, @Nullable Object o4) {",
+            "    Utilities.takesNonNullVarargs(o1, o2, o3);",
+            "    Utilities.takesNonNullVarargs(o1);", // Empty var args passed
+            "    // BUG: Diagnostic contains: passing @Nullable parameter 'o4' where @NonNull",
+            "    Utilities.takesNonNullVarargs(o1, o4);",
+            "  }",
+            "}")
+        .doTest();
+  }
+
+  @Test
+  public void testNullableVarargs() {
+    makeHelper()
+        .addSourceLines(
+            "Utilities.java",
+            "package com.uber;",
+            "import javax.annotation.Nullable;",
+            "public class Utilities {",
+            " public static String takesNullableVarargs(Object o, @Nullable Object... others) {",
+            "  String s = o.toString() + \" \" + others.toString();",
+            "  return s;",
```

Comment:
Shouldn't we be checking in here that the elements _are_ `@Nullable`? By adding the for loop and trying to deref `other? Or is the table saying that in JSpecify mode (and every mode other than legacy) we don't check declaration annotations (but acknowledge them for call sites)?

Reply:
I did add the test in bc46f2e448887afa05f03cb879a403faf2013325.  And I have now changed the logic so that we get an error, in 6b622c7d855c9464d599bf90eefa6a9426d79af0.  Great catch!  My table didn't have this case as an error before, due to an attempt to make treatment of array types identical for regular arrays and varargs arrays.  But there are already other cases where we don't have identical treatment.  And the previous table made a `@Nullable` declaration annotation a no-op within the method body, which could be confusing.  So this is now fixed, and I will update the table accordingly.